airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/common.py

```diff
@@ -0,0 +1,51 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from dataclasses import dataclass, field
+from typing import List
+
+from pydantic import BaseModel
+
+from .schemas.profile import Profile
+
+URL_BASE = "https://advertising-api.amazon.com/"
+
+
+class Config(BaseModel):
+    client_id: str
+    client_secret: str
+    scope: str
+    refresh_token: str
+
+
+@dataclass
+class PageToken:
+    offset: int = 0
+
+
+@dataclass
```

Comment:
can we use single tools for a class generation? i.e either pydantic or data classes, what are you missing in pydantic?

Reply:
pydantic is for json serialization/deserialization and dataclasses is for making structured classes. Using pydantic is overkill here because we do not need deserialization from json.